### ISSUE TITLE
代码问题

### ISSUE THREAD

**Issue by Cjl-MedSeg** [workforai/UniLSeg#5]
if not self.cfg.aux_loss:
          pred = torch.bmm(query_output, pixel_output.flatten(2)) 
          pred = rearrange(pred, 'b l (h w) -> b l h w', h=h, w=w)   
      else:
          for l, q in enumerate(query_output):
              final_output = []
              pred = torch.bmm(query_output[l], pixel_output.flatten(2))
              pred = rearrange(pred, 'b l (h w) -> b l h w', h=h, w=w)
              final_output.append(pred)
      return pred.detach()
请问这里为什么会输出最后pred，其他5个pred得作用是什么呢?
麻烦您解决我的困惑，非常感谢！！！！

**Comment by yongliu20** [comment]
这里是因为在训练时对中间层的pred也施加loss约束，类似DETR的auxiliary losses。最终inference时都是取最后一层的pred进行mask输出
